### PR TITLE
Improve strings formatting: NetworkStrings and OfferingStrings

### PR DESCRIPTION
**Pull request by Juanpe**
Follow-up on #816 with changes on `NetworkStrings` and `OfferingStrings`

Blocked by #817 and #818

BTW, I've found some duplicate messages related to **StoreKit**. Might it make sense to create a `StoreKitStrings` to group all these messages? 

In addition, I've found some unused tokens:
```
OffertinsStrings.retrieved_products
OffertinsStrings.list_products
OffertinsStrings.invalid_product_identifiers
OffertinsStrings.fetching_products_finished
OffertinsStrings.fetching_products
OffertinsStrings.completion_handlers_waiting_on_products

NetworkStrings.could_not_find_cached_response
```

Could we remove them? 🤔 

### Requirements: 
- [ ] #818

### CODE REVIEW

File: Purchases/Logging/Strings/OfferingStrings.swift [hunk 1]
```diff
@@ -13,35 +13,121 @@
...
+
+    case skproductsrequest_did_finish
+
+    case skproductsrequest_received_response
+
+    case vending_offerings_cache
+
+    case retrieved_products(products: [SKProduct])
+
+    case list_products(inputOne: String, inputTwo: String)
```
Comment (Juanpe) [inline Purchases/Logging/Strings/OfferingStrings.swift:49]: this key is not used and I couldn't determine what both inputs mean 😕

Comment (aboedo) [reply Purchases/Logging/Strings/OfferingStrings.swift:49]: This log entry was lost during the migration: https://github.com/RevenueCat/purchases-ios/blob/d0ac5b9b422207f7c710d04cb77899e4e77d9495/Purchases/Purchasing/RCStoreKitRequestFetcher.m

During the migration, the storekit request management was split into products requests and refresh receipt requests. 
The corresponding method for this log is now found here https://github.com/RevenueCat/purchases-ios/blob/swift_migration/Purchases/Purchasing/ProductsManager.swift#L72, and it's not using our logging 🤦 
This is my bad, the file was there from before the migration and it predates some of our logging stuff, and I forgot to integrate it afterwards. 

I think we should just have the params with the correct names (which would be `productIdentifier: String` and `product: SKProduct`, and then update the `ProductsManager` file to add the correct logging. I'm happy to update the `ProductsManager` as a part of #800

Comment (aboedo) [reply Purchases/Logging/Strings/OfferingStrings.swift:49]: the other unused strings are probably related to that as well

Comment (Juanpe) [reply Purchases/Logging/Strings/OfferingStrings.swift:49]: Ok, I'm going to update the case then :) thanks for the clarification 👍🏼

**Comment by aboedo** [comment]
> BTW, I've found some duplicate messages related to StoreKit. Might it make sense to create a StoreKitStrings to group all these messages?
I'm all for this

**Comment by Juanpe** [comment]
Great! I could include StoreKitString in this PR, WDYT?

**Comment by aboedo** [comment]
sounds great!

**Comment by Juanpe** [comment]
I've created the new `enum` to group all the strings related to **StoreKit**. I have checked all the strings and didn't find any more. Let me know if you know or find some **StoreKit** related strings to include them in this `enum` too 🙂

File: Purchases/Logging/Strings/StoreKitStrings.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKitStrings.swift
+//
+//  Created by Juanpe Catalán on 8/9/21.
+
+import Foundation
+
+// swiftlint:disable identifier_name
+enum StoreKitStrings {
```
Comment (aboedo) [inline Purchases/Logging/Strings/StoreKitStrings.swift:17]: ❤️

**Review by aboedo — APPROVED** [review]
Amazing work as usual @Juanpe!! 
We can add more stuff to `StoreKitStrings` when we come across something that looks very storekit-specific. Thanks so much for doing this! 

I updated the base to #818 since this PR seems to be based on that one, and we'll merge it as soon as that one gets merged.

**Comment by Juanpe** [comment]
> I updated the base to #818 since this PR seems to be based on that one, and we'll merge it as soon as that one gets merged. 

Ok, it makes sense 👍🏼

Thanks one more time for the review and your feedback ❤️

**Comment by aboedo** [comment]
@Juanpe I merged the `improve_strings_formatting_2_codable_strings` and github auto-closed this!!! 
I'm so sorry about that. It must have something to do with the original branch coming from a fork, because when I do that on most branches, it usually auto-updates the base branch to main (after the base branch was merged). 
I apparently can't re-open, could you re-open (or a new one if GH doesn't let you) and I can merge it? 

Again, I'm so sorry, I never expected GitHub to behave that way

**Comment by Juanpe** [comment]
No problem @aboedo. TBH, I did it a lot of times and it always works as you said. So, I agree with you that it must have something to do with the original branch coming from a fork. In that case, good to know 😅 I'll open a new PR with the changes because I can't re-open this one, but I have a doubt...considering #828, what should be the base branch, `main` or `swift_migration`?

**Comment by aboedo** [comment]
Just merged the other one, you can base off of `main`!

**Pull request closed without merging**